tools/swift-format/Sources/Rules/UseTripleSlashForDocumentationComments.swift

```diff
@@ -4,17 +4,72 @@ import SwiftSyntax
 
 /// Documentation comments must use the `///` form.
 ///
-/// Flag comments (e.g. `// TODO(username):`) are exempted from this rule.
+/// This is similar to `NoBlockComments` but is meant to prevent documentation block comments.
 ///
-/// This is similar to `NoBlockComments` but is meant to prevent multi-line comments that use `//`.
+/// Lint: If a doc block comment appears, a lint error is raised.
 ///
-/// Lint: If a declaration has a multi-line comment preceding it and that comment is not in `///`
-///       form, a lint error is raised.
-///
-/// Format: If a declaration has a multi-line comment preceding it and that comment is not in `///`
-///         form, it is converted to the `///` form.
+/// Format: If a doc block comment appears on its own on a line, or if a doc block comment spans multiple
+///         lines without appearing on the same line as code, it will be replaced with multiple
+///         doc line comments.
 ///
 /// - SeeAlso: https://google.github.io/swift#general-format
-public final class UseTripleSlashForDocumentationComments {
+public final class UseTripleSlashForDocumentationComments: SyntaxFormatRule {
+  public override func visit (_ token: TokenSyntax) -> Syntax {
+    var pieces = [TriviaPiece]()
+    var isInvalid = false
+    
+    // Ensures that all doc block comments are replaced with doc line comment,
+    // unless the comment is between tokens on the same line.
+    for piece in token.leadingTrivia {
+      if case .docBlockComment(let text) = piece,
+         !commentIsBetweenCode(token) {
+        isInvalid = true
+        diagnose(.avoidDocBlockComment, on: token)
+        let docLineCommentText = convertsDocBlockCommentToDocLineComment(text)
+        let docLineComment = TriviaPiece.docLineComment(docLineCommentText)
+        pieces.append(docLineComment)
+      }
+      else {
+        pieces.append(piece)
+      }
+    }
+    return isInvalid ? token.withLeadingTrivia(Trivia.init(pieces: pieces)) : token
+  }
+  
+  /// Indicates if a doc block comment is between tokens on the same line.
```

Comment:
Since this is duplicated from `NoBlockComments`, it might be best to move this into a convenience method inside `TokenSyntax+Convenience.swift`